doc/sphinx/src/using-kpt.rst

```diff
@@ -0,0 +1,248 @@
+(see equation for Gibbs free energy above). Equilibrium phase transition mass fraction tables will be made available through ``singularity-eos`` in the near future.
+
+Kinetic phase transition models
+'''''''''''''''''''''''''''''''
+
+We write the new mass fractions at time :math:`t` as
+
+.. math::
+
+ \mu_i^{t} = \mu_i^{t_0} + \frac{d\mu_i}{dt} dt  
+
+which with discretized time becomes
+
+.. math::
+
+ \mu_i^{t} = \mu_i^{t_0} + \dot{\mu_i} \Delta t
+
+Using mass conservation (:math:`\sum_i \mu_i = 1`), we see that 
+
+.. math::
+
+ 0 = \sum_i \dot{\mu_i} = \sum_i \sum_j ( \mu_j R_{ji} - \mu_i R_{ij} )
+   
+where :math:`R_{ij}` is the mass transportation rate from phase :math:`i` to phase :math:`j`, and we can derive the master equation
+
+.. math::
+
+ \dot{\mu_i} = \sum_j ( \mu_j R_{ji} - \mu_i R_{ij} )
+
+This equation simply states the fact that all mass transforming from one phase ends up in another phase, it is
+just mass conservation.
+
+A KPT model is a model for the :math:`R_{ij}`.
+
+Carl Greeff's KPT model
+'''''''''''''''''''''''
+
+Carl Greeff formulated an empirical mass fraction update model as
+
+.. math::
+
+ R_{ij} = \nu_{ij} \theta(G_i-G_j) \frac{G_i-G_j}{B_{ij}} \exp\left[ \left( \frac{G_i-G_j}{B_{ij}}\right)^2 \right]
+
+where :math:`\nu_{ij}` and :math:`B_{ij}` are material dependent fitting constants. 
+Mattsson-Wills performed a `parameter study <WillsParameterstudy_>`_ of this model, 
+which can be used as a guide for how to choose these parameters for a specific material and a specific phase transition.
+
+This model is included in ``singularity-eos`` and its signature is
+
+.. code-block:: cpp
+
+  LogRatesCGModel(const Real *w, const Real *b, const int num_phases, const Real *gibbs,
+                  const int *gibbsorder, Real *logRij, int *fromto)
+
+where ``w`` is :math:`\nu_{ij}`, ``b`` is :math:`B_{ij}`, ``num_phases`` is
+:math:`N`, the number of phases, ``gibbs`` is :math:`G_i`, ``gibbsorder`` is an array of length :math:`N`, where the ``gibbs`` phase indices are ordered 
+from the largest Gibbs free energy phase to the lowest Gibbs free energy phase (see figure below),
+``logRij`` is :math:`\log(R_{ij})`, and ``fromto`` is a map between the phase indices :math:`i` and :math:`j` and the :math:`ij` phase transition indices in :math:`R_{ij}`. 
+The ``gibbs`` and ``gibbsorder`` arrays are of length :math:`N` while ``w`` and ``b`` are arrays of length :math:`N^2` representing the :math:`N \times N` matrices, row by row. 
+Note that it is *NOT* assumed that the phase transition parameters are the same when going from :math:`i \rightarrow j` and  :math:`j \rightarrow i`, that is, :math:`\{\nu,B\}_{ij} \neq \{\nu,B\}_{ji}`.
+Also note that :math:`R_{ii} = 0`.
+``logRij`` is an array of length :math:`N (N-1)/2` giving the logarithm of the non-zero mass transportation rates between phases. The ``fromto`` array 
```

Comment:
I decided not to, since I have not really introduced any matrices. But it is easy to see that if $R_{ij}$ is non-zero, $R_{ji}$ would be zero (because of the Heaviside step function, oh, I forgot to say that that is what it is....). Maybe I can say that. With $R_{ii}=0$ this means a triangle of size N*(N-1)/2. But there are many way to come to the same number...

Reply:
Done.